server/http/controller.go

```diff
@@ -1221,3 +1225,92 @@ func (c *controller) TestConnection(ctx context.Context, dataStore openapi.DataS
 
 	return openapi.Response(statusCode, c.mappers.Out.ConnectionTestResult(testResult)), nil
 }
+
+func (c *controller) GetTestVariables(ctx context.Context, testId, environmentId string) (openapi.ImplResponse, error) {
+	test, err := c.testDB.GetLatestTestVersion(ctx, id.ID(testId))
+	if err != nil {
+		return handleDBError(err), err
+	}
+
+	environment := model.Environment{
+		Values: []model.EnvironmentValue{},
+	}
+
+	if environmentId != "" {
+		environment, err = c.testDB.GetEnvironment(ctx, environmentId)
+
+		if err != nil {
+			return handleDBError(err), err
+		}
+	}
+
+	executor := expression.NewExecutor()
+	variables := expression.NewVariables(environment, executor)
+
+	testVariables, err := processTestVariables(ctx, test, variables, c.triggerRegistry)
+
+	if err != nil {
+		return openapi.Response(http.StatusInternalServerError, err.Error()), err
+	}
+
+	return openapi.Response(200, c.mappers.Out.TestVariables(testVariables)), nil
+}
+
+func processTestVariables(ctx context.Context, test model.Test, variables expression.Variables, triggerRegistry *trigger.Registry) (expression.TestVariables, error) {
+	environmentVariables := variables.GetEnvironmentVariables(test)
+	specVariables, err := variables.GetSpecsVariables(test)
```

Comment:
Do we need a similar step for `Outputs`? Similar to specs we can reference a `env:` var in an output value, right?

Reply:
You are right, I'm adding it

Reply:
Done 😄